[GStreamer][WebRTC] RTP header extension ID handling improvements for SDP Offers/Answers

philn opened the pull request:
#### d9547193f4ce867409e1285333ae9c24359c3b1c
<pre>
[GStreamer][WebRTC] RTP header extension ID handling improvements for SDP Offers/Answers
<a href="https://bugs.webkit.org/show_bug.cgi?id=303094">https://bugs.webkit.org/show_bug.cgi?id=303094</a>

Reviewed by Xabier Rodriguez-Calvar.

Our SDP offers and answers were advertizing RTP header extensions incorrectly, where the same
extension URL was using different numerical identifiers on each media section of the SDP. The
mapping is now consistent across media sections and comparable to SDP produced by Chrome. We also
now emit an error if too many extensions are requested.

Canonical link: <a href="https://commits.webkit.org/303677@main">https://commits.webkit.org/303677@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2c8ddfc5b4836ed3613ed0a400ebe42345fac26d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/133272 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/5773 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/44404 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/140826 "Built successfully") | [❌ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/85318 "Failed to checkout and rebase branch from PR 54546") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/135142 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/6282 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/5638 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/101944 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/59/builds/85318 "Failed to checkout and rebase branch from PR 54546") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/136219 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/4466 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/119459 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/82740 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/4350 "Passed tests") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/1927 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [⏳ 🛠 wpe-cairo ](https://ews-build.webkit.org/#/builders/WPE-Cairo-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/113429 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/37574 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/143474 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/5443 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/38151 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/110321 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/5525 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/4679 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/110505 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28001 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/4214 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/115713 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/59193 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/5498 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/34067 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/5344 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/68950 "Built successfully") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/5587 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/5454 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

calvaris commented on Source/WebCore/Modules/mediastream/gstreamer/GStreamerMediaEndpoint.cpp:
```diff
@@ -1620,6 +1620,22 @@ ExceptionOr<GStreamerMediaEndpoint::Backends> GStreamerMediaEndpoint::createTran
+        // signaling section, this is referred to as the valid range.
+        auto identifier = m_rtpHeaderExtensions.size() + 1;
+        ASSERT(identifier <= 14);
+        if (identifier > 14) {
```
You can put this inside the if as an ASSERT_NOT_REACHED[_WITH_MESSAGE]

philn committed:
[GStreamer][WebRTC] RTP header extension ID handling improvements for SDP Offers/Answers
https://bugs.webkit.org/show_bug.cgi?id=303094

Reviewed by Xabier Rodriguez-Calvar.

Our SDP offers and answers were advertizing RTP header extensions incorrectly, where the same
extension URL was using different numerical identifiers on each media section of the SDP. The
mapping is now consistent across media sections and comparable to SDP produced by Chrome. We also
now emit an error if too many extensions are requested.

Canonical link: https://commits.webkit.org/303677@main
